src/tasks/sync-commands/datasources.js

```diff
@@ -13,13 +13,31 @@ class SyncDatasources {
     this.targetSpaceId = options.targetSpaceId
     this.oauthToken = options.oauthToken
     this.client = api.getClient()
+    this.datasourcesStartsWithSlug = options.datasourcesStartsWithSlug
+    this.datasourcesStartsWithName = options.datasourcesStartsWithName
   }
 
   async sync () {
     try {
       this.targetDatasources = await this.client.getAll(`spaces/${this.targetSpaceId}/datasources`)
       this.sourceDatasources = await this.client.getAll(`spaces/${this.sourceSpaceId}/datasources`)
 
+      if (this.datasourcesStartsWithSlug) {
+        this.sourceDatasources = this.sourceDatasources.filter(datasource => datasource.slug.toLowerCase().startsWith(this.datasourcesStartsWithSlug.toLowerCase()));
+        const filteredSlugs = this.sourceDatasources.map(obj => obj.slug);
+        const formattedSlugs = filteredSlugs.join(', ');
+
+        console.log(`${chalk.blue('-')} Source datasources where slug starts with ${this.datasourcesStartsWithSlug}: ${formattedSlugs}`);
+      }
+  
+      if (this.datasourcesStartsWithName) {
+        this.sourceDatasources = this.sourceDatasources.filter(datasource => datasource.name.toLowerCase().startsWith(this.datasourcesStartsWithName.toLowerCase()));
+        const filteredNames = this.sourceDatasources.map(obj => obj.name);
+        const formattedNames = filteredNames.join(', ');
+
+        console.log(`${chalk.blue('-')} Source datasources where name starts with ${this.datasourcesStartsWithName}: ${formattedNames}`);
```

Comment:
That part could be extracted into a separate function, as the same piece of code is repeated twice.